### PR TITLE
Removed `pallet::getter` usage from Polkadot Runtime pallets

### PR DESCRIPTION
**Pull request by muraca**
Part of #3326 

@kianenigma @ggwpez 

polkadot address: 12poSUQPtcF1HUPQGY3zZu2P8emuW9YnsPduA4XG3oCEfJVp

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5505102

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5505553

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5506738

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5564812

File: polkadot/runtime/parachains/src/disputes.rs [hunk 1]
```diff
@@ -853,6 +852,14 @@ impl StatementSetFilter {
...
 
 impl<T: Config> Pallet<T> {
+	/// Whether the chain is frozen. Starts as `None`. When this is `Some`,
+	/// the chain will not accept any new parachain blocks for backing or inclusion,
+	/// and its value indicates the last valid block number in the chain.
+	/// It can only be set back to `None` by governance intervention.
+	pub fn last_valid_block() -> Option<BlockNumberFor<T>> {
+		Frozen::<T>::get()
+	}
+
```
Comment (liamaharon) [inline polkadot/runtime/parachains/src/disputes.rs:862]: Why's this needed?

Comment (muraca) [reply polkadot/runtime/parachains/src/disputes.rs:862]: It is not, sorry. I'll remove it and set the `StorageValue` as `pub` as it was done for the other items

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5567089

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5579627

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5629164

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5629466

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5632222

File: polkadot/runtime/parachains/src/hrmp.rs [hunk 2]
```diff
@@ -1018,8 +1018,8 @@ impl<T: Config> Pallet<T> {
 			let recipient_deposit = if system_channel { 0 } else { config.hrmp_recipient_deposit };
 
 			if request.confirmed {
-				if <paras::Pallet<T>>::is_valid_para(channel_id.sender) &&
-					<paras::Pallet<T>>::is_valid_para(channel_id.recipient)
+				if paras::Pallet::<T>::is_valid_para(channel_id.sender) &&
```
Comment (kianenigma) [inline polkadot/runtime/parachains/src/hrmp.rs:1021]: If you ever get to explore https://github.com/trailofbits/dylint for our preferred syntax, I'd be very curious to hear about it :)

**Review by kianenigma — APPROVED** [review]
A bit of a painful review, but did it :D If need be, could have been broken down into smaller PRs. 

@ordian @eskimor can you also do a review, or at least approve the direction, as it affects a lot of your code base.

**Review by ordian — APPROVED** [review]
Thanks for the ping @kianenigma. I'm fine with this change assuming this has no effect on visibility of chain state on polkadot js, is that right?
Screenshot 2024-04-01 at 21 50 16

File: polkadot/runtime/parachains/src/configuration.rs [hunk 3]
```diff
@@ -512,8 +512,7 @@ pub mod pallet {
 	/// The active configuration for the current session.
 	#[pallet::storage]
 	#[pallet::whitelist_storage]
-	#[pallet::getter(fn config)]
-	pub(crate) type ActiveConfig<T: Config> =
+	pub type ActiveConfig<T: Config> =
```
[the remark this answers is not in the document]
Comment (ordian) [reply polkadot/runtime/parachains/src/configuration.rs:515]: I'm guessing previously we could specify different visibility for getters and setters and we'd lose this ability now.

**Comment by kianenigma** [comment]
> Thanks for the ping @kianenigma. I'm fine with this change assuming this has no effect on visibility of chain state on polkadot js, is that right? Screenshot 2024-04-01 at 21 50 16

Yes, removing the getters is a 100% noop in the metadata and the external API.

**Review by ggwpez — APPROVED** [review]

File: polkadot/runtime/parachains/src/configuration.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (ggwpez) [reply polkadot/runtime/parachains/src/configuration.rs:515]: I think getters are always pub.

**Comment by kianenigma** [comment]
@muraca ready for last upstream sync and then we can merge.

File: polkadot/runtime/parachains/src/configuration.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (muraca) [reply polkadot/runtime/parachains/src/configuration.rs:515]: Getters are always pub indeed, this change is required as it is the only way we currently have, other than writing manually all getters.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5810744

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5810743

**Comment by muraca** [comment]
tip?

**Pull request merged**